tests/pytests/unit/states/test_beacon.py

```diff
@@ -0,0 +1,64 @@
+"""
+    :codeauthor: :email:`Jayesh Kariya <jayeshk@saltstack.com>`
+"""
+
+import pytest
+import salt.states.beacon as beacon
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {beacon: {}}
+
+
+def test_present():
+    """
+    Test to ensure a job is present in the beacon.
+    """
+    beacon_name = "ps"
+
+    ret = {"name": beacon_name, "changes": {}, "result": False, "comment": ""}
+
+    mock_mod = MagicMock(return_value=ret)
+    mock_lst = MagicMock(side_effect=[{beacon_name: {}}, {beacon_name: {}}, {}, {}])
+    with patch.dict(
+        beacon.__salt__,
+        {
+            "beacons.list": mock_lst,
+            "beacons.modify": mock_mod,
+            "beacons.add": mock_mod,
+        },
+    ):
+        assert beacon.present(beacon_name) == ret
+
+        with patch.dict(beacon.__opts__, {"test": False}):
+            assert beacon.present(beacon_name) == ret
+
+            assert beacon.present(beacon_name) == ret
```

Comment:
```suggestion
            assert beacon.present(beacon_name) == ret
```
If the two calls are required, we should add a comment telling why

Reply:
I don't see any reason to call same asserts twice, perhaps it's a typo in original test case

Reply:
Fixed